0-module-and-module-source.emu

```diff
@@ -0,0 +1,614 @@
+              1. Let _importEntries_ be _sourceModule_.[[ImportEntries]].
+              1. Let _localExportEntries_ be _sourceModule_.[[LocalExportEntries]].
+              1. Let _indirectExportEntries_ be _sourceModule_.[[IndirectExportEntries]].
+              1. Let _starExportEntries_ be _sourceModule_.[[StarExportEntries]].
+              1. Let _hostDefined_ be _sourceModule_.[[HostDefined]].
+              1. Let _realm_ be the current Realm Record.
+              1. Return Source Text Module Record { [[Realm]]: _realm_, [[Environment]]: ~empty~, [[Namespace]]: ~empty~, [[CycleRoot]]: ~empty~, [[HasTLA]]: _async_, [[AsyncEvaluation]]: *false*, [[TopLevelCapability]]: ~empty~, [[AsyncParentModules]]: &laquo; &raquo;, [[PendingAsyncDependencies]]: ~empty~, [[Status]]: ~unlinked~, [[EvaluationError]]: ~empty~, [[HostDefined]]: _hostDefined_, [[ECMAScriptCode]]: _body_, [[Context]]: ~empty~, [[ImportMeta]]: ~empty~, [[RequestedModules]]: _requestedModules_, [[ImportEntries]]: _importEntries_, [[LocalExportEntries]]: _localExportEntries_, [[IndirectExportEntries]]: _indirectExportEntries_, [[StarExportEntries]]: _starExportEntries_, [[DFSIndex]]: ~empty~, [[DFSAncestorIndex]]: ~empty~, [[ModuleInstance]]: *undefined* }.
+            </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-resolvemodulerecorddependency" type="abstract operation">
+            <h1>
+                ResolveModuleRecordDependency (
+                    _moduleRecord_: a Module Record,
+                    _specifier_: a |ModuleSpecifier| String,
+                    _promiseCapability_: a PromiseCapability Record,
+                ): ~unused~
+            </h1>
+            <dl class="header">
+              <dt>description</dt>
+              <dd>TODO: It provides the concrete Module Record subclass instance that corresponds to _specifier_ occurring within the module represented by _moduleRecord_.</dd>
+            </dl>
+            <emu-alg>
+              1. Assert: _moduleRecord_ is a Module Record.
+              1. Assert: _moduleRecord_.[[ModuleInstance]] is an Object.
+              1. Assert: ResolveModuleRecordDependency has not been invoked with _moduleRecord_ and _specifier_ pair.
+              1. Let _moduleInstance_ be _moduleRecord_.[[ModuleInstance]].
+              1. Let _importHook_ be _moduleInstance_.[[ImportHook]].
+              1. Let _completion_ be Completion(Call(_importHook_, *undefined*, « _specifier_ »).
+              1. IfAbruptRejectPromise(_completion_, _promiseCapability_).
+              1. Let _importHookPromise_ be ? PromiseResolve(%Promise%, _completion_.[[Value]]).
+              1. Let _fulfilledClosure_ be a new Abstract Closure with parameters (_result_) that captures _moduleRecord_ and _promiseCapability_ and performs the following steps when called
+                1. Assert: Evaluate has already been invoked on _moduleRecord_ and successfully completed.
```

Comment:
this does not look correct to me, it will prevent load nested dependencies.

Reply:
+1, Evaluate cannot have been called if we are still loading the module dependencies.